Sources/JSONSchemaBuilder/JSONComponent/TypeSpecific/JSONObject.swift

```diff
@@ -1,39 +1,51 @@
 import JSONSchema
 
 /// A JSON object schema component for use in ``JSONSchemaBuilder``.
-public struct JSONObject: JSONSchemaComponent {
+public struct JSONObject<each Property: JSONPropertyComponent>: JSONSchemaComponent {
   public var annotations: AnnotationOptions = .annotations()
-  var options: ObjectSchemaOptions = .options()
+  var options: ObjectSchemaOptions
 
-  public var definition: Schema { .object(annotations, options) }
-
-  public init() {}
```

Comment:
Follow-up: Need to think more on adding back empty JSONObject and JSONArray inits.